### ISSUE TITLE
$push and ReadonlyArray<T> does not play nice

### ISSUE THREAD

**Issue by m-rutter** [kolodny/immutability-helper#154]
I am having issues with `$push` and `ReadonlyArray` types, which is very common in graphql codegen generated interfaces.

## Minimal Example
```ts
interface Thing {
  bar: {
    foo: ReadonlyArray<{ baz: number }>;
  };
}

declare const a: Thing;
declare const b: Thing;

update(a, { bar: { foo: { $push: b.bar.foo } } });


```

returns this type error:

```
Type 'readonly { baz: number; }[]' is not assignable to type 'Spec<{ baz: number; }, never>'.
  Type 'readonly { baz: number; }[]' is not assignable to type '(v: { baz: number; }) => { baz: number; }'.
    Type 'readonly { baz: number; }[]' provides no match for the signature '(v: { baz: number; }): { baz: number; }'.ts(2322)
```

## Workaround

Use `$set` instead

**Comment by kolodny** [comment]
Good catch, opened #155 to address this

**Comment by kolodny** [comment]
Published to 3.0.2

**Comment by donatocayurin** [comment]
> Good catch, opened #155 to address this

thank you very much sir
